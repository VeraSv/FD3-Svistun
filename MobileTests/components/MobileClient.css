.Active {
  background-color: chartreuse;
}

.Blocked {
  background-color: coral;
}