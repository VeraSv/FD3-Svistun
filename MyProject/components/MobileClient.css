div.MobileClient
{
  padding: 2px 0;
}

span.MobileClientBalance
{
  float: right;
  width: 30px;
  font-weight: bold;
  text-align: right;
}

.MobileClientFIO
{
  margin-right: 30px;
}
