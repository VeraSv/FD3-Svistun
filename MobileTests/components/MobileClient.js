import React from 'react';
import PropTypes from 'prop-types';
import {events} from './events';
import './MobileClient.css';

class MobileClient extends React.PureComponent {

  static propTypes = {
client: PropTypes.object
 
  };

  edit=(EO)=>{
    events.emit('EditClicked',this.props.client.id);
  }

  delete=(EO)=>{
    events.emit('DeleteClicked',this.props.client.id);
  }
  render() {
    console.log("MobileClient id="+this.props.client.id+" render");
    return (
     <tr key={this.props.client.name} >
         
      <td>{this.props.client.surname}</td> 
      <td>{this.props.client.name}</td> 
      <td>{this.props.client.middleName}</td>
      <td>{this.props.client.balance}</td>
    <td className={(this.props.client.balance>0)?'Active':'Blocked'}>{(this.props.client.balance>0)?'Active':'Blocked'}</td>
        <td>
        <input type={'button'} value={'Редактировать'} id='Edit' onClick={this.edit}/></td>
        <td> <input type={'button'} value={'Удалить'}  id='Delete' onClick={this.delete}/> </td>
     </tr>
    );

  }

}

export default MobileClient;
