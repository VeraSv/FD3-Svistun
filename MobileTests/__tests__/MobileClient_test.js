"use strict";

import React from 'react';
import renderer from 'react-test-renderer';

import MobileClient from '../components/MobileClient.js';

test('работа MobileClient', () => {
  let client={id:1,name:'Иван', surname:'Иванов', middleName:'Иванович', balance:200}
    
    
    const component = renderer.create(
        <MobileClient client={client} />
      );

      let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const buttonEdit = component.root.find((el) =>
  el.type == 'input' && el.props.className == 'Edit');
  buttonEdit.props.onClick();

  const buttonDelete = component.root.find((el) =>
  el.type == 'input' && el.props.className == 'Delete');
  buttonDelete.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  buttonEdit.props.onClick();
  buttonDelete.props.onClick();
 

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

    
})