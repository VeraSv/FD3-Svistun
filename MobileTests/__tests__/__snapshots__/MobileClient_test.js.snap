// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа MobileClient 1`] = `
<tr>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    className="Active"
  >
    Active
  </td>
  <td>
    <input
      className="Edit"
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
     
    <input
      className="Delete"
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
     
  </td>
</tr>
`;

exports[`работа MobileClient 2`] = `
<tr>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    className="Active"
  >
    Active
  </td>
  <td>
    <input
      className="Edit"
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
     
    <input
      className="Delete"
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
     
  </td>
</tr>
`;

exports[`работа MobileClient 3`] = `
<tr>
  <td>
    Иванов
  </td>
  <td>
    Иван
  </td>
  <td>
    Иванович
  </td>
  <td>
    200
  </td>
  <td
    className="Active"
  >
    Active
  </td>
  <td>
    <input
      className="Edit"
      onClick={[Function]}
      type="button"
      value="Редактировать"
    />
  </td>
  <td>
     
    <input
      className="Delete"
      onClick={[Function]}
      type="button"
      value="Удалить"
    />
     
  </td>
</tr>
`;
